t/TestDNS.pm

```diff
@@ -91,13 +94,37 @@ sub gen_dns_reply ($$) {
         $answers = [$answers];
     }
 
+    my $authority_answers = $t->{authority} // [];
+    if (!ref $authority_answers) {
+	$authority_answers = [$authority_answers];	
```

Comment:
Style: please remove the line trailing spaces on this line and possibly other lines in your patch.